aws/aws-sdk-java-v2
# Enable JDK9+ code and builds to produce Java8 compatible jar

## Motivation and Context

* Newly developed Java9 code written with module-info.java/.class should still produce jars that run compatibly on JRE8 as this SDK is still required to support it.



## Modifications

* Maven profile enabled JDK9+ build also triggers a separate java-8-compile excluding module-info.java
* maven-compiler-plugin executions are in the parent pom
* maven-jar-plugin needed to be at least 3.1.2 to enable ModuleMainClass in tested submodule not included in PR so this is an unnecessary change
* module-path-tests adjusted to suppress java-8-compile execution.

## Testing

1. Added new test sub-module under core/<module> set up with module-info.java and main method
1. Verify build succeeds when JAVA_HOME and PATH points to JDK8 and JDK9+ (tested with 17)
1. Execute main method through JRE8 classpath successfully
1. Execute main method through JRE8 with module-path successfully



*  Changes impacted module-path-tests submodule. Suppressed java-8-compile execution
## Screenshots (if appropriate)

## Types of changes


## Checklist



## License
